astroid/brain/brain_attrs.py

```diff
@@ -0,0 +1,38 @@
+"""Astroid hook for the attrs library
+
+Without this hook pylint reports unsupported-assignment-operation
+for atrrs classes
+"""
+import astroid
+from astroid import MANAGER
+
+
+def register(linter):
+  # Needed for registering the plugin.
+  pass
+
+
+def _attr_assign_attributes(node):
+    for cdefbodynode in node.body:
+        if not isinstance(cdefbodynode, astroid.Assign):
+            continue
+        for target in cdefbodynode.targets:
+            node.locals[target.name] = [astroid.Attribute(target.name)]
+
+
+def attr_attributes_transform(node):
+    """If the given ClassNode has an attr decorator,
+    then rewrite class attributes as instance attributes
+    """
+    if not node.decorators:
+        return None
+    for decorator_node in node.decorators.nodes:
+        if not isinstance(decorator_node, astroid.Attribute):
+            continue
+        if decorator_node.as_string() in ("attr.s",
+                                          "attr.attributes",
+                                          "attr.attrs"):
+            _attr_assign_attributes(node)
+
+
+MANAGER.register_transform(astroid.Class, attr_attributes_transform)
```

Comment:
We should pass the filter from the transform here instead. Something along the lines of:

```
MANAGER.register_transform(astroid.Class, attr_attributes_transform, is_decorated_with_attrs)
```
where `is_decorated_with_attrs` returns True or False